ITKImageProcessingFilters/itkInPlaceDream3DDataToImageFilter.h

```diff
@@ -0,0 +1,57 @@
+#ifndef _itkInPlaceDream3DDataToImageFilter_h
+#define _itkInPlaceDream3DDataToImageFilter_h
+
+#include <itkImportImageFilter.h>
+
+namespace itk
+{
+  template< typename PixelType, unsigned int VDimension >
+  class InPlaceDream3DDataToImageFilter : public ImageSource< itk::Image<PixelType, VDimension> >
+  {
+  public:
+    /** Standard class typedefs. */
+    typedef InPlaceDream3DDataToImageFilter                                 Self;
+    typedef SmartPointer<Self>                                              Pointer;
+    typedef typename itk::Image<PixelType, VDimension>                      ImageType;
+    typedef typename ImageType::Pointer                                     ImagePointer;
+    typedef typename ::DataArray<PixelType>                                 DataArrayPixelType;
+    typedef typename itk::ImageSource< itk::Image<PixelType, VDimension> >  Superclass;
+    typedef itk::SizeValueType                                              SizeValueType;
+    typedef ImportImageContainer< SizeValueType, PixelType >                ImportImageContainerType;
+    typedef Image< PixelType, VDimension >                                  OutputImageType;
+    typedef typename OutputImageType::Pointer                               OutputImagePointer;
+
+    /** Method for creation through the object factory. */
+    itkNewMacro(Self);
+    itkTypeMacro(InPlaceDream3DDataToImageFilter, ImportImageFilter);
+    virtual void SetInput(DataContainer::Pointer &dc);
+    itkSetMacro(DataArrayName, std::string);
+    itkGetMacro(DataArrayName, std::string);
```

Comment:
Can you also add API to take a DataArrayPath object as a convenience method to set the AttributeMatrixArrayName and the DataArrayName in one method.